### PR TITLE
fix(marketo): add check for rETL events during getting leadId

### PR DESCRIPTION
**Pull request by ItsSudip**
## Description of the change

[INT-134] Small fix for marketo

## Type of change
- [x] Bug fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Related issues

> Fix [#1]() 

### Development

- [ ] Lint rules pass locally
- [ ] The code changed/added as part of this pull request has been covered with tests
- [ ] All tests related to the changed code pass in development

### Code review 

- [ ]  This pull request has a descriptive title and information useful to a reviewer. There may be a screenshot or screencast attached
- [ ] "Ready for review" label attached to the PR and reviewers mentioned in a comment
- [ ] Changes have been reviewed by at least one other engineer
- [ ] Issue from task tracker has a link to this pull request

### CODE REVIEW

File: src/cdk/v1/heap/mapping/track.yaml [hunk 1]
```diff
@@ -18,8 +18,9 @@
     type: flatJson
 
 - destKey: timestamp
-  sourceKeys: timestamp
-  sourceFromGenericMap: true
+  sourceKeys:
+    - timestamp
+    - originalTimestamp
```
Comment (yashasvibajpai) [inline src/cdk/v1/heap/mapping/track.yaml:23]: Can we please mention these heap changes too in description if they are valid changes?

File: test/__tests__/data/tiktok_ads_input.json [hunk 2]
```diff
@@ -2264,7 +2264,7 @@
         "eventId": "1616318632825_357",
         "products": [
           {
-            "product_id": "123",
+            "product_id": 123,
```
Comment (yashasvibajpai) [inline test/__tests__/data/tiktok_ads_input.json:2267]: Is it fine to also convert other product related parameters like content type or category to string too? Does tiktok work fine with numberical values/ structures like object in its place?

**Review by koladilip — APPROVED** [review]

File: src/cdk/v1/heap/mapping/track.yaml (hunk 1 above)
Comment (ItsSudip) [reply src/cdk/v1/heap/mapping/track.yaml]: @yashasvibajpai These changes are coming as the develop was not up to date with main. After updating this change is not part of this commit.

File: test/__tests__/data/tiktok_ads_input.json (hunk 2 above)
Comment (ItsSudip) [reply test/__tests__/data/tiktok_ads_input.json]: same here

**Review by yashasvibajpai — APPROVED** [review]

**Review by sanpj2292 — APPROVED** [review]

**Pull request merged**